manifests/0000_90_etcd-operator_03_prometheusrule.yaml

```diff
@@ -130,6 +119,17 @@ spec:
         severity: warning
   - name: openshift-etcd.rules
     rules:
+    - alert: etcdGRPCRequestsSlow
+      annotations:
+        description: 'etcd cluster "{{ $labels.%s }}": 99th percentile of gRPC requests is {{ $value }}s on etcd instance {{ $labels.instance }} for {{ $labels.grpc_method }} method.'
+        runbook_url: https://github.com/openshift/runbooks/blob/master/alerts/cluster-etcd-operator/etcdGRPCRequestsSlow.md
+        summary: etcd grpc requests are slow
+      expr: |
+        histogram_quantile(0.99, sum(rate(grpc_server_handling_seconds_bucket{job="etcd", grpc_method!="Defragment", grpc_type="unary"}[10m])) without(grpc_type))
+        > 1
+      for: 30m
+      labels:
+        severity: critical
```

Comment:
`critical` is [midnight-page territory][1].  And currently the only direct mitigation mentioned in the runbook is [running a defrag][2].  Perhaps the alert should pivot from "right now things are awful!" to "the past day has been pretty weak", and it could be a `warning` so folks could start thinking about provisioning faster disks in the near future?  Because it's hard for me to imaging rolling out of bed to run a defrag, or to scale up my control plane disks, and feeling like that was a healthy UX.

[1]: https://github.com/openshift/enhancements/blob/15aebc7549095cfc7eebc26d6460defc8e3927f6/enhancements/monitoring/alerting-consistency.md#critical-alerts
[2]: https://github.com/openshift/runbooks/blob/8513f248783bceddc3280e9387f7ac68e0ed0985/alerts/cluster-etcd-operator/etcdGRPCRequestsSlow.md#mitigation

Reply:
as mentioned above, I'm going to also update the runbook. Will send you the link to the updated one later today.
I would definitely want to get woken up if etcd takes longer than 1s to respond to anything. I'll revise the mitigation. 

> so folks could start thinking about provisioning faster disks

On BM that makes sense, in clouds you can easily scale up your VM type to unlock more IOPS or a faster CPU in a couple of minutes.

Reply:
https://github.com/openshift/runbooks/pull/69

Reply:
> I would definitely want to get woken up if etcd takes longer than 1s to respond to anything.

Would you like to have had a `warning` ping the day before that response times were frequently up over 0.25s?  Or would that be too noisy, and you'd rather not hear about that and take the midnight page if/when it gets up to 1s?

Reply:
I would, if we assume that this is something that degrades slowly over time. The failures we observe in CI are transient, they come briefly and go away again. Would you pro-actively change anything in the build cluster when you see such warning the day before? 

Maybe we can find some folks over at SD to see whether their ROSA/ARO clusters would benefit from such warning. Not sure they would find that actionable as such and it would just drown in the rest of their 400 alerts a day.